mycroft/configuration/config.py

```diff
@@ -203,9 +203,18 @@ def load_config_stack(configs=None, cache=False):
         Returns:
             (dict) merged dict of all configuration files
         """
+        # system administrators can define different constraints in how
+        # configurations are loaded
+        system_conf = LocalConf(SYSTEM_CONFIG).get("system") or {}
+        protected_keys = system_conf.get("protected_keys") or {}
+        protected_remote = protected_keys.get("remote") or []
+        protected_user = protected_keys.get("user") or []
+        skip_user = system_conf.get("disable_user_config", False)
+        skip_remote = system_conf.get("disable_remote_config", False)
+
```

Comment:
We could also have a `user_configurable_keys` which would act as the inverse of `protected_keys`.
ie if a project wants to protect everything except x and y:
```
"user_configurable_keys": ["x", "y"]
```

This is just an idea though, not necessary in v1 of this. I think let's add it if we see actual need for it.